### PR TITLE
Order variables reported by xcube server

### LINKED ISSUE #835: `xcube server`: specify variable to be displayed by default in xcube viewer
**Is your feature request related to a problem? Please describe.**
When I publish a dataset via xcube server which is then visualised by xcube viewer, the variables are variables in the drop down menu are sorted alphabetically. And per default the first variable in the list is visualised. However, this is not always an variable of interest for the user of xcube viewer. 

**Describe the solution you'd like**
I would like to be able to specify a variable which should be used per default in xcube viewer. To achieve this, I would like to be able to specify it in the xcube server configuration yaml file or also be able to indicate the default variable already in the datasets metadata. 

**Describe alternatives you've considered**
Currently there is no alternative to this. 

**Additional context**

### PR DESCRIPTION
**Pull request by forman**
It is now possible to enforce the order of variables reported by xcube server. The new server configuration key `Variables` can be added to dataset configurations. Is a list of wildcard patterns that determine the variables' order.

Closes #835

Checklist:

* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in `docs/source/*`
* [ ] Changes documented in `CHANGES.md`
* [ ] AppVeyor CI passes
* [ ] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -3,6 +3,11 @@
 * Setting a dataset's `BoundingBox` in the server configuration 
   is now recognised when requesting the dataset details. (#845)
 
+* It is now possible to enforce the order of variables reported by 
+  xcube server. The new server configuration key `Variables` can be added 
+  to dataset configurations. Is a list of wildcard patterns that 
+  determine the variables' order. (#835)
```
Comment (AliceBalfanz) [inline CHANGES.md:9]: Please add, that if for a dataset only a subset of variables is defined and **not** followed by `- "*"`, then only the specified variables are displayed.

**Review by AliceBalfanz — CHANGES_REQUESTED** [review]
Please see comment. Furthermore, I am missing an update in the documentation. But this should not be blocking, I can create a PR for that.

File: CHANGES.md (hunk 1 above)
Comment (forman) [reply CHANGES.md:9]: Good point!

**Comment by forman** [comment]
> Please see comment. Furthermore, I am missing an update in the documentation. But this should not be blocking, I can create a PR for that.

I'll do that too.

**Review by AliceBalfanz — APPROVED** [review]
thank you for adding the documentation!

**Pull request merged**